texlive: 2023-final -> 2024.20241027

veprbl opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Resolves: #350953

Update to 2024 snapshot release

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

veprbl committed:
texlive: 2023-final -> 2024.20241027

veprbl commented on pkgs/tools/typesetting/tex/texlive/tlpdb-overrides.nix:
```diff
@@ -126,6 +126,11 @@ in lib.recursiveUpdate orig rec {
     "mtxrun.lua" = tl.context.tex + "/scripts/context/lua/mtxrun.lua";
   };
 
+  context-legacy.binlinks = {
```
I'm not sure why, but "ctxtools" and "pstopdf" are not in `context-legacy.binfiles`. I wasn't sure if we are supposed to still provide them or not.

mirkolenz commented:
Thanks for the speedy PR! I just applied texlive 2024 to my nix-darwin config and it worked without any issues. I can also confirm that my issue with `acro` is fixed with these changes.

I tried to run `nixpkgs-review` for this PR, but get the following error:

<details>
<summary>nixpkgs-review pr 351790</summary>

```txt
error:
… while querying the derivation named 'manim-0.18.1'

… while calling the 'getAttr' builtin
  at <nix/derivation-internal.nix>:44:19:
    43|       value = commonAttrs // {
    44|         outPath = builtins.getAttr outputName strict;
      |                   ^
    45|         drvPath = strict.drvPath;

… while calling the 'derivationStrict' builtin
  at <nix/derivation-internal.nix>:34:12:
    33|
    34|   strict = derivationStrict drvAttrs;
      |            ^
    35|

… while evaluating derivation 'python3.12-manim-0.18.1'
  whose name attribute is located at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

… while evaluating attribute '__impureHostDeps' of derivation 'python3.12-manim-0.18.1'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:451:7:
  450|       __propagatedSandboxProfile = unique (computedPropagatedSandboxProfile ++ [ propagatedSandboxProfile ]);
  451|       __impureHostDeps = computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ stdenv.__extraImpureHostDeps ++ [
      |       ^
  452|         "/dev/zero"

… while calling the 'foldl'' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:11:
  434|         computedImpureHostDeps =
  435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
      |           ^
  436|             (stdenv.extraNativeBuildInputs

… while calling the 'concatMap' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:19:
  434|         computedImpureHostDeps =
  435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
      |                   ^
  436|             (stdenv.extraNativeBuildInputs

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:30:
  434|         computedImpureHostDeps =
  435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
      |                              ^
  436|             (stdenv.extraNativeBuildInputs

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:37:
  434|         computedImpureHostDeps =
  435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
      |                                     ^
  436|             (stdenv.extraNativeBuildInputs

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:307:13:
  306|       (map (drv: getDev drv.__spliced.buildBuild or drv) (checkDependencyList "depsBuildBuild" depsBuildBuild))
  307|       (map (drv: getDev drv.__spliced.buildHost or drv) (checkDependencyList "nativeBuildInputs" nativeBuildInputs'))
      |             ^
  308|       (map (drv: getDev drv.__spliced.buildTarget or drv) (checkDependencyList "depsBuildTarget" depsBuildTarget))

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:307:18:
  306|       (map (drv: getDev drv.__spliced.buildBuild or drv) (checkDependencyList "depsBuildBuild" depsBuildBuild))
  307|       (map (drv: getDev drv.__spliced.buildHost or drv) (checkDependencyList "nativeBuildInputs" nativeBuildInputs'))
      |                  ^
  308|       (map (drv: getDev drv.__spliced.buildTarget or drv) (checkDependencyList "depsBuildTarget" depsBuildTarget))

… while calling 'getOutput'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1797:23:
  1796|   */
  1797|   getOutput = output: pkg:
      |                       ^
  1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified

… while evaluating a branch condition
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1798:5:
  1797|   getOutput = output: pkg:
  1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
      |     ^
  1799|       then pkg.${output} or pkg.out or pkg

… in the left operand of the OR (||) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1798:32:
  1797|   getOutput = output: pkg:
  1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
      |                                ^
  1799|       then pkg.${output} or pkg.out or pkg

… in the argument of the not operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1798:10:
  1797|   getOutput = output: pkg:
  1798|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
      |          ^
  1799|       then pkg.${output} or pkg.out or pkg

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:307:25:
  306|       (map (drv: getDev drv.__spliced.buildBuild or drv) (checkDependencyList "depsBuildBuild" depsBuildBuild))
  307|       (map (drv: getDev drv.__spliced.buildHost or drv) (checkDependencyList "nativeBuildInputs" nativeBuildInputs'))
      |                         ^
  308|       (map (drv: getDev drv.__spliced.buildTarget or drv) (checkDependencyList "depsBuildTarget" depsBuildTarget))

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/lists.nix:334:29:
  333|   */
  334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
      |                             ^
  335|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/lists.nix:334:32:
  333|   */
  334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
      |                                ^
  335|

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:281:15:
  280|     imap1
  281|       (index: dep:
      |               ^
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep

… while evaluating a branch condition
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:9:
  281|       (index: dep:
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
      |         ^
  283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

… in the left operand of the OR (||) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:69:
  281|       (index: dep:
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
      |                                                                     ^
  283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

… in the left operand of the OR (||) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:44:
  281|       (index: dep:
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
      |                                            ^
  283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

… in the left operand of the OR (||) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:29:
  281|       (index: dep:
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
      |                             ^
  283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:12:
  281|       (index: dep:
  282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
      |            ^
  283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

… while calling 'isDerivation'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1283:5:
  1282|   isDerivation =
  1283|     value: value.type or null == "derivation";
      |     ^
  1284|

… while calling the 'elemAt' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/lists.nix:334:43:
  333|   */
  334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
      |                                           ^
  335|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/development/python-modules/manim/default.nix:55:19:
    54|   # these two combined add up to:
    55|   manim-tinytex = texliveInfraOnly.withPackages (
      |                   ^
    56|     ps: with ps; [

… while calling 'withPackages'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:213:20:
  212|         self (args // { __fromCombineWrapper = false; } // appliedArgs);
  213|     withPackages = reqs: self (args // { requiredTeXPackages = ps: reqs ps ++ requiredTeXPackages ps; __fromCombineWrapper = false; });
      |                    ^
  214|   };

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:213:26:
  212|         self (args // { __fromCombineWrapper = false; } // appliedArgs);
  213|     withPackages = reqs: self (args // { requiredTeXPackages = ps: reqs ps ++ requiredTeXPackages ps; __fromCombineWrapper = false; });
      |                          ^
  214|   };

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:26:16:
    25|
    26| lib.fix (self: {
      |                ^
    27|   withDocs ? false

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:315:54:
  314|   # outputsToInstall must be set *after* overrideAttrs (used in buildEnv') or it fails the checkMeta tests
  315| in if __combine || __formatsOf != null then out else lib.addMetaAttrs { inherit (pkgList) outputsToInstall; } out)
      |                                                      ^
  316|

… while calling 'addMetaAttrs'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/meta.nix:42:28:
    41|   */
    42|   addMetaAttrs = newAttrs: drv:
      |                            ^
    43|     drv // { meta = (drv.meta or {}) // newAttrs; };

… in the left operand of the update (//) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/meta.nix:43:9:
    42|   addMetaAttrs = newAttrs: drv:
    43|     drv // { meta = (drv.meta or {}) // newAttrs; };
      |         ^
    44|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:260:1:
  259| # no indent for git diff purposes
  260| buildEnv' {
      | ^
  261|

… while calling 'buildEnv''
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:44:15:
    43|   ### buildEnv with custom attributes
    44|   buildEnv' = args: (buildEnv
      |               ^
    45|     ({ inherit (args) name paths; })

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:44:21:
    43|   ### buildEnv with custom attributes
    44|   buildEnv' = args: (buildEnv
      |                     ^
    45|     ({ inherit (args) name paths; })

… while calling 'overrideAttrs'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:152:73:
  151|           overrideDerivation = fdrv: overrideResult (x: overrideDerivation x fdrv);
  152|           ${if result ? overrideAttrs then "overrideAttrs" else null} = fdrv:
      |                                                                         ^
  153|             overrideResult (x: x.overrideAttrs fdrv);

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:153:13:
  152|           ${if result ? overrideAttrs then "overrideAttrs" else null} = fdrv:
  153|             overrideResult (x: x.overrideAttrs fdrv);
      |             ^
  154|         }

… while calling 'overrideResult'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:146:24:
  145|       # Change the result of the function call by applying g to it
  146|       overrideResult = g: makeOverridable (mirrorArgs (args: g (f args))) origArgs;
      |                        ^
  147|     in

… while calling a functor (an attribute set with a '__functor' attribute)
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:146:27:
  145|       # Change the result of the function call by applying g to it
  146|       overrideResult = g: makeOverridable (mirrorArgs (args: g (f args))) origArgs;
      |                           ^
  147|     in

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/trivial.nix:966:7:
  965|     { # TODO: Should we add call-time "type" checking like built in?
  966|       __functor = self: f;
      |       ^
  967|       __functionArgs = args;

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:136:17:
  135|     in
  136|     mirrorArgs (origArgs:
      |                 ^
  137|     let

… while evaluating a branch condition
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:148:7:
  147|     in
  148|       if isAttrs result then
      |       ^
  149|         result // {

… while calling the 'isAttrs' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:148:10:
  147|     in
  148|       if isAttrs result then
      |          ^
  149|         result // {

… while calling a functor (an attribute set with a '__functor' attribute)
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:138:16:
  137|     let
  138|       result = f origArgs;
      |                ^
  139|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/trivial.nix:966:7:
  965|     { # TODO: Should we add call-time "type" checking like built in?
  966|       __functor = self: f;
      |       ^
  967|       __functionArgs = args;

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:146:56:
  145|       # Change the result of the function call by applying g to it
  146|       overrideResult = g: makeOverridable (mirrorArgs (args: g (f args))) origArgs;
      |                                                        ^
  147|     in

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:146:62:
  145|       # Change the result of the function call by applying g to it
  146|       overrideResult = g: makeOverridable (mirrorArgs (args: g (f args))) origArgs;
      |                                                              ^
  147|     in

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:153:29:
  152|           ${if result ? overrideAttrs then "overrideAttrs" else null} = fdrv:
  153|             overrideResult (x: x.overrideAttrs fdrv);
      |                             ^
  154|         }

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:153:32:
  152|           ${if result ? overrideAttrs then "overrideAttrs" else null} = fdrv:
  153|             overrideResult (x: x.overrideAttrs fdrv);
      |                                ^
  154|         }

… while calling anonymous lambda
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:100:8:
    99|     mkDerivationSimple
  100|       (f0:
      |        ^
  101|         let

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:110:11:
  109|         in
  110|           makeDerivationExtensible (self: attrs // (if builtins.isFunction f0 || f0?__functor then f self attrs else f0)))
      |           ^
  111|       attrs;

… while calling 'makeDerivationExtensible'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:77:30:
    76|   # Based off lib.makeExtensible, with modifications:
    77|   makeDerivationExtensible = rattrs:
      |                              ^
    78|     let

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:92:9:
    91|       finalPackage =
    92|         mkDerivationSimple overrideAttrs args;
      |         ^
    93|

… while calling 'mkDerivationSimple'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:516:1:
  515| # * [package attributes]: https://nixos.org/manual/nix/stable/glossary#package-attribute-set
  516| {
      | ^
  517|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:596:1:
  595|
  596| extendDerivation
      | ^
  597|   validity.handled

… while calling 'extendDerivation'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:339:43:
  338|   */
  339|   extendDerivation = condition: passthru: drv:
      |                                           ^
  340|     let

… in the left operand of the update (//) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:364:20:
  363|       outputsList = map outputToAttrListElement outputs;
  364|     in commonAttrs // {
      |                    ^
  365|       drvPath = assert condition; drv.drvPath;

… in the left operand of the update (//) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/customisation.nix:343:25:
  342|
  343|       commonAttrs = drv // (listToAttrs outputsList) //
      |                         ^
  344|         ({ all = map (x: x.value) outputsList; }) // passthru;

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:645:4:
  644|    passthru)
  645|   (derivation (derivationArg // optionalAttrs envIsExportable checkedEnv));
      |    ^
  646|

… while calling anonymous lambda
  at <nix/derivation-internal.nix>:30:1:
    29| */
    30| drvAttrs @ { outputs ? [ "out" ], ... }:
      | ^
    31|

… while calling the 'head' builtin
  at <nix/derivation-internal.nix>:53:5:
    52|
    53| in (builtins.head outputsList).value
      |     ^
    54|

… while calling the 'map' builtin
  at <nix/derivation-internal.nix>:51:17:
    50|
    51|   outputsList = map outputToAttrListElement outputs;
      |                 ^
    52|

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:265:26:
  264|   # use attrNames, attrValues to ensure the two lists are sorted in the same way
  265|   outputs = [ "out" ] ++ lib.optionals (! __combine && __formatsOf == null) (builtins.attrNames nonEnvOutputs);
      |                          ^
  266|   otherOutputs = lib.optionals (! __combine && __formatsOf == null) (builtins.attrValues nonEnvOutputs);

… while calling 'optionals'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/lists.nix:820:5:
  819|     cond:
  820|     elems: if cond then elems else [];
      |     ^
  821|

… while calling the 'attrNames' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:265:78:
  264|   # use attrNames, attrValues to ensure the two lists are sorted in the same way
  265|   outputs = [ "out" ] ++ lib.optionals (! __combine && __formatsOf == null) (builtins.attrNames nonEnvOutputs);
      |                                                                              ^
  266|   otherOutputs = lib.optionals (! __combine && __formatsOf == null) (builtins.attrValues nonEnvOutputs);

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:190:19:
  189|   # other outputs
  190|   nonEnvOutputs = lib.genAttrs pkgList.nonEnvOutputs (outName: buildEnv' {
      |                   ^
  191|     inherit name;

… while calling 'genAttrs'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1247:5:
  1246|     names:
  1247|     f:
      |     ^
  1248|     listToAttrs (map (n: nameValuePair n (f n)) names);

… while calling the 'listToAttrs' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1248:5:
  1247|     f:
  1248|     listToAttrs (map (n: nameValuePair n (f n)) names);
      |     ^
  1249|

… while calling the 'map' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/lib/attrsets.nix:1248:18:
  1247|     f:
  1248|     listToAttrs (map (n: nameValuePair n (f n)) names);
      |                  ^
  1249|

… while evaluating the attribute 'nonEnvOutputs'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:112:5:
  111|     # outputs that do not become part of the environment
  112|     nonEnvOutputs = lib.subtractLists [ "out" "tex" "texdoc" "texsource" "tlpkg" ] otherOutputNames;
      |     ^
  113|

… while calling the 'filter' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:112:21:
  111|     # outputs that do not become part of the environment
  112|     nonEnvOutputs = lib.subtractLists [ "out" "tex" "texdoc" "texsource" "tlpkg" ] otherOutputNames;
      |                     ^
  113|

… while calling the 'catAttrs' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:81:24:
    80|     specifiedOutputs = lib.groupBy (p: p.tlOutputName or p.outputName) specified.right;
    81|     otherOutputNames = builtins.catAttrs "key" (builtins.genericClosure {
      |                        ^
    82|       startSet = map (key: { inherit key; }) (lib.concatLists (builtins.catAttrs "outputs" specified.wrong));

… while calling the 'genericClosure' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:81:49:
    80|     specifiedOutputs = lib.groupBy (p: p.tlOutputName or p.outputName) specified.right;
    81|     otherOutputNames = builtins.catAttrs "key" (builtins.genericClosure {
      |                                                 ^
    82|       startSet = map (key: { inherit key; }) (lib.concatLists (builtins.catAttrs "outputs" specified.wrong));

… while calling the 'map' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:82:18:
    81|     otherOutputNames = builtins.catAttrs "key" (builtins.genericClosure {
    82|       startSet = map (key: { inherit key; }) (lib.concatLists (builtins.catAttrs "outputs" specified.wrong));
      |                  ^
    83|       operator = _: [ ];

… while calling the 'concatLists' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:82:47:
    81|     otherOutputNames = builtins.catAttrs "key" (builtins.genericClosure {
    82|       startSet = map (key: { inherit key; }) (lib.concatLists (builtins.catAttrs "outputs" specified.wrong));
      |                                               ^
    83|       operator = _: [ ];

… while calling the 'catAttrs' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:82:64:
    81|     otherOutputNames = builtins.catAttrs "key" (builtins.genericClosure {
    82|       startSet = map (key: { inherit key; }) (lib.concatLists (builtins.catAttrs "outputs" specified.wrong));
      |                                                                ^
    83|       operator = _: [ ];

… while calling the 'partition' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:79:17:
    78|     # group the specified outputs
    79|     specified = builtins.partition (p: p.outputSpecified or false) all;
      |                 ^
    80|     specifiedOutputs = lib.groupBy (p: p.tlOutputName or p.outputName) specified.right;

… while calling the 'catAttrs' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:73:9:
    72|       if __fromCombineWrapper then requiredTeXPackages null else
    73|         builtins.catAttrs "p" (builtins.genericClosure {
      |         ^
    74|           startSet = map keySet runtime.right;

… while calling the 'genericClosure' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:73:32:
    72|       if __fromCombineWrapper then requiredTeXPackages null else
    73|         builtins.catAttrs "p" (builtins.genericClosure {
      |                                ^
    74|           startSet = map keySet runtime.right;

… while calling the 'map' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:74:22:
    73|         builtins.catAttrs "p" (builtins.genericClosure {
    74|           startSet = map keySet runtime.right;
      |                      ^
    75|           operator = p: map keySet p.tlDeps;

… while calling the 'partition' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:62:19:
    61|         packages = ensurePkgSets (requiredTeXPackages tl);
    62|         runtime = builtins.partition
      |                   ^
    63|           (p: p.outputSpecified or false -> builtins.elem (p.tlOutputName or p.outputName) [ "out" "tex" "tlpkg" ])

… from call site
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:61:20:
    60|       let
    61|         packages = ensurePkgSets (requiredTeXPackages tl);
      |                    ^
    62|         runtime = builtins.partition

… while calling 'ensurePkgSets'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:52:19:
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps
      |                   ^
    53|     then let oldPkgLists = builtins.partition isOldPkgList ps;

… while evaluating a branch condition
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:52:23:
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps
      |                       ^
    53|     then let oldPkgLists = builtins.partition isOldPkgList ps;

… in the right operand of the AND (&&) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:52:49:
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps
      |                                                 ^
    53|     then let oldPkgLists = builtins.partition isOldPkgList ps;

… while calling the 'any' builtin
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:52:52:
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps
      |                                                    ^
    53|     then let oldPkgLists = builtins.partition isOldPkgList ps;

… while calling 'isOldPkgList'
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:51:18:
    50|   # if necessary, convert old style { pkgs = [ ... ]; } packages to attribute sets
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
      |                  ^
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps

… in the left operand of the AND (&&) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:51:62:
    50|   # if necessary, convert old style { pkgs = [ ... ]; } packages to attribute sets
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
      |                                                              ^
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps

… in the left operand of the AND (&&) operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:51:50:
    50|   # if necessary, convert old style { pkgs = [ ... ]; } packages to attribute sets
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
      |                                                  ^
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps

… in the argument of the not operator
  at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/tools/typesetting/tex/texlive/build-tex-env.nix:51:23:
    50|   # if necessary, convert old style { pkgs = [ ... ]; } packages to attribute sets
    51|   isOldPkgList = p: ! p.outputSpecified or false && p ? pkgs && builtins.all (p: p ? tlType) p.pkgs;
      |                       ^
    52|   ensurePkgSets = ps: if ! __fromCombineWrapper && builtins.any isOldPkgList ps

error: undefined variable 'ms'
at ~/.cache/nixpkgs-review/pr-351790-2/nixpkgs/pkgs/development/python-modules/manim/default.nix:158:7:
  157|       doublestroke
  158|       ms
      |       ^
  159|       setspac
```

</details>

veprbl commented:
@mirkolenz May be an unlucky state of master. I rebased again, hoping eval will pass now.

mirkolenz commented:
> May be an unlucky state of master. I rebased again, hoping eval will pass now.

Still got the same error with nixpkgs-review. But I agree that this is most likely some upstream issue not related to this PR since I was able to build it as part of my system config just fine.

veprbl committed:
python312Packages.manim: don't depend on texlive.ms

According to https://new.ctan.org/pkg/ms it was split into other
packages.

According to
https://github.com/3b1b/manim/blob/master/manimlib/tex_templates.yml
only ragged2e is used (already depended on).

xworld21 commented:
> Still got the same error with nixpkgs-review. But I agree that this is most likely some upstream issue not related to this PR since I was able to build it as part of my system config just fine.

Apparently the `ms` package is no more https://ctan.org/pkg/ms. One should make manim depend on the relevant subset of the packages listed there.

veprbl commented:
@GrahamcOfBorg build tests.texlive

xworld21 commented on pkgs/tools/typesetting/tex/texlive/bin.nix:
```diff
@@ -40,10 +40,10 @@ let
+              "ftp://tug.ctan.org/pub/tex/historic/systems/texlive/${year}/texlive-${year}0312-source.tar.xz"
       ];
-      hash = "sha256-OHiqDh7QMBwFOw4u5OmtmZxEE0X0iC55vdHI9M6eebk=";
+      hash = "sha256-e22HzwFmFnD6xFyTEmvtl7mEMTntUQ+XXQR+qTi2/pY=";
```
```nix
src = fetchsvn {
  url = "svn://tug.org/texlive/branches/branch2024/Build/source";
  rev = 70897;
  hash = "sha256-xSoi26/UJ9gQw7nherMNR6upYZdrwOwW/lMSPQSRTk0=";
};
```
Better yet with url `svn://tug.org/texlive/tags/texlive-2024.2/Build/source`, revision 70951 but you'd have to check if the hash is the same (see https://tug.org/svn/texlive/tags/texlive-2024.2/Build/source/). This is necessary to include:
- https://tug.org/pipermail/tlbuild/2024q1/005519.html
- https://tug.org/pipermail/tlbuild/2024q2/005529.html

xworld21 replied:
PS: there should be a copy on the TeX Live GitHub clone, e.g. tag svn70897, if we prefer GitHub as mirror.

veprbl replied:
This is now implemented (via `fetchFromGitHub`). Thanks for the pointers!

mirkolenz commented:
Great investigation of the issue 😄 

I tried running `nixpkgs-review` with the latest changes and it went fine for about 3 hours or so, but eventually my computer ran out of memory and was barely usable (I have 32 GB available but that was not enough), so I canceled the run. It seems that many Python packages needed to be rebuilt.

veprbl commented:
> I tried running `nixpkgs-review` with the latest changes and it went fine for about 3 hours or so, but eventually my computer ran out of memory and was barely usable (I have 32 GB available but that was not enough), so I canceled the run. It seems that many Python packages needed to be rebuilt.

If that's on NixOS, you may want to adjust number of parallel jobs and cores per job because the defaults may be to aggressive for massive builds:
```nix
  nix.settings = {
    cores = 2;
    max-jobs = 16;
  };
```
(an example for a 32-core machine).

veprbl commented:
Result of `nixpkgs-review pr 351790` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages marked as broken and skipped:</summary>
  <ul>
    <li>frescobaldi</li>
    <li>frescobaldi.dist</li>
    <li>hid-tools</li>
    <li>hid-tools.dist</li>
    <li>labplot</li>
    <li>manim</li>
    <li>manim-slides</li>
    <li>manim-slides.dist</li>
    <li>manim.dist</li>
    <li>mate.atril</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.manim</li>
    <li>python311Packages.manim-slides</li>
    <li>python311Packages.manim-slides.dist</li>
    <li>python311Packages.manim.dist</li>
    <li>python312Packages.manim</li>
    <li>python312Packages.manim-slides</li>
    <li>python312Packages.manim-slides.dist</li>
    <li>python312Packages.manim.dist</li>
    <li>quarto</li>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
    <li>yosys-bluespec</li>
  </ul>
</details>
<details>
  <summary>59 packages failed to build:</summary>
  <ul>
    <li>aider-chat</li>
    <li>aider-chat.dist</li>
    <li>aligator</li>
    <li>aligator.doc</li>
    <li>bonmin</li>
    <li>casadi</li>
    <li>crocoddyl</li>
    <li>crocoddyl.doc</li>
    <li>herwig</li>
    <li>pinocchio</li>
    <li>pinocchio.doc</li>
    <li>proxsuite-nlp</li>
    <li>proxsuite-nlp.doc</li>
    <li>pynac</li>
    <li>python311Packages.aerosandbox</li>
    <li>python311Packages.aerosandbox.dist</li>
    <li>python311Packages.aligator</li>
    <li>python311Packages.aligator.doc</li>
    <li>python311Packages.casadi</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.crocoddyl.doc</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.example-robot-data.doc</li>
    <li>python311Packages.neuralfoil</li>
    <li>python311Packages.neuralfoil.dist</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.pinocchio.doc</li>
    <li>python311Packages.proxsuite-nlp</li>
    <li>python311Packages.proxsuite-nlp.doc</li>
    <li>python311Packages.recoll</li>
    <li>python311Packages.tsid</li>
    <li>python311Packages.tsid.doc</li>
    <li>python311Packages.unstructured</li>
    <li>python311Packages.unstructured.dist</li>
    <li>python312Packages.aerosandbox</li>
    <li>python312Packages.aerosandbox.dist</li>
    <li>python312Packages.aligator</li>
    <li>python312Packages.aligator.doc</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.crocoddyl.doc</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.example-robot-data.doc</li>
    <li>python312Packages.neuralfoil</li>
    <li>python312Packages.neuralfoil.dist</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.pinocchio.doc</li>
    <li>python312Packages.proxsuite-nlp</li>
    <li>python312Packages.proxsuite-nlp.doc</li>
    <li>recoll (python312Packages.recoll)</li>
    <li>python312Packages.tsid</li>
    <li>python312Packages.tsid.doc</li>
    <li>recoll-nox</li>
    <li>sherpa</li>
    <li>sympow</li>
    <li>thepeg</li>
    <li>tsid</li>
    <li>tsid.doc</li>
    <li>vimPlugins.aider-nvim</li>
  </ul>
</details>
<details>
  <summary>176 packages built:</summary>
  <ul>
    <li>R</li>
    <li>R.tex</li>
    <li>advi</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>asymptote</li>
    <li>asymptote.doc</li>
    <li>asymptote.info</li>
    <li>asymptote.man</li>
    <li>asymptote.tex</li>
    <li>auctex</li>
    <li>auctex.tex</li>
    <li>auto-multiple-choice</li>
    <li>biber</li>
    <li>biber.devdoc</li>
    <li>blahtexml</li>
    <li>blahtexml.doc</li>
    <li>cddlib</li>
    <li>dblatex</li>
    <li>dblatexFull</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>dot2tex (python312Packages.dot2tex)</li>
    <li>dot2tex.dist (python312Packages.dot2tex.dist)</li>
    <li>eclib</li>
    <li>evince</li>
    <li>evince.dev</li>
    <li>evince.devdoc</li>
    <li>extractpdfmark</li>
    <li>gap</li>
    <li>gap-full</li>
    <li>gap-minimal</li>
    <li>gfan</li>
    <li>giac</li>
    <li>giac.doc</li>
    <li>gnucobol</li>
    <li>gnucobol.bin</li>
    <li>gnucobol.dev</li>
    <li>gnucobol.lib</li>
    <li>gp2c</li>
    <li>inkscape-extensions.textext</li>
    <li>inkscape-extensions.textext.dist</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>latte-integrale</li>
    <li>lcalc</li>
    <li>lilypond</li>
    <li>lilypond-unstable</li>
    <li>lilypond-unstable-with-fonts</li>
    <li>lilypond-with-fonts</li>
    <li>ne</li>
    <li>nuweb</li>
    <li>pari</li>
    <li>postgresql12JitPackages.plr</li>
    <li>postgresql12Packages.plr</li>
    <li>postgresql13JitPackages.plr</li>
    <li>postgresql13Packages.plr</li>
    <li>postgresql14JitPackages.plr</li>
    <li>postgresql14Packages.plr</li>
    <li>postgresql15JitPackages.plr</li>
    <li>postgresql15Packages.plr</li>
    <li>postgresqlJitPackages.plr (postgresql16JitPackages.plr)</li>
    <li>postgresqlPackages.plr (postgresql16Packages.plr)</li>
    <li>postgresql17JitPackages.plr</li>
    <li>postgresql17Packages.plr</li>
    <li>python311Packages.abjad</li>
    <li>python311Packages.abjad.dist</li>
    <li>python311Packages.cnvkit</li>
    <li>python311Packages.cnvkit.dist</li>
    <li>python311Packages.cypari2</li>
    <li>python311Packages.cypari2.dist</li>
    <li>python311Packages.cysignals</li>
    <li>python311Packages.cysignals.dist</li>
    <li>python311Packages.dot2tex</li>
    <li>python311Packages.dot2tex.dist</li>
    <li>python311Packages.fpylll</li>
    <li>python311Packages.fpylll.dist</li>
    <li>python311Packages.pplpy</li>
    <li>python311Packages.pplpy.dist</li>
    <li>python311Packages.pplpy.doc</li>
    <li>python311Packages.primecountpy</li>
    <li>python311Packages.primecountpy.dist</li>
    <li>python311Packages.pylatex</li>
    <li>python311Packages.pylatex.dist</li>
    <li>python311Packages.pypandoc</li>
    <li>python311Packages.pypandoc.dist</li>
    <li>python311Packages.pytikz-allefeld</li>
    <li>python311Packages.pytikz-allefeld.dist</li>
    <li>python311Packages.radian</li>
    <li>python311Packages.radian.dist</li>
    <li>python311Packages.rchitect</li>
    <li>python311Packages.rchitect.dist</li>
    <li>python311Packages.rivet</li>
    <li>python311Packages.rpy2</li>
    <li>python311Packages.rpy2.dist</li>
    <li>python311Packages.sphinxcontrib-bayesnet</li>
    <li>python311Packages.sphinxcontrib-bayesnet.dist</li>
    <li>python311Packages.sphinxcontrib-tikz</li>
    <li>python311Packages.sphinxcontrib-tikz.dist</li>
    <li>python312Packages.cnvkit</li>
    <li>python312Packages.cnvkit.dist</li>
    <li>python312Packages.cypari2</li>
    <li>python312Packages.cypari2.dist</li>
    <li>python312Packages.cysignals</li>
    <li>python312Packages.cysignals.dist</li>
    <li>python312Packages.fpylll</li>
    <li>python312Packages.fpylll.dist</li>
    <li>python312Packages.pplpy</li>
    <li>python312Packages.pplpy.dist</li>
    <li>python312Packages.pplpy.doc</li>
    <li>python312Packages.primecountpy</li>
    <li>python312Packages.primecountpy.dist</li>
    <li>python312Packages.pylatex</li>
    <li>python312Packages.pylatex.dist</li>
    <li>python312Packages.pypandoc</li>
    <li>python312Packages.pypandoc.dist</li>
    <li>python312Packages.pytikz-allefeld</li>
    <li>python312Packages.pytikz-allefeld.dist</li>
    <li>python312Packages.radian</li>
    <li>python312Packages.radian.dist</li>
    <li>python312Packages.rchitect</li>
    <li>python312Packages.rchitect.dist</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>python312Packages.rpy2</li>
    <li>python312Packages.rpy2.dist</li>
    <li>python312Packages.sphinxcontrib-bayesnet</li>
    <li>python312Packages.sphinxcontrib-bayesnet.dist</li>
    <li>python312Packages.sphinxcontrib-tikz</li>
    <li>python312Packages.sphinxcontrib-tikz.dist</li>
    <li>python312Packages.unstructured</li>
    <li>python312Packages.unstructured.dist</li>
    <li>resorter</li>
    <li>sagetex.tex</li>
    <li>singular</li>
    <li>skribilo</li>
    <li>spark (spark3 ,spark_3_5)</li>
    <li>spark_3_4</li>
    <li>spip</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libR</li>
    <li>texliveBasic</li>
    <li>texliveBasic.info</li>
    <li>texliveBasic.man</li>
    <li>texliveBookPub</li>
    <li>texliveBookPub.info</li>
    <li>texliveBookPub.man</li>
    <li>texliveConTeXt</li>
    <li>texliveConTeXt.info</li>
    <li>texliveConTeXt.man</li>
    <li>texliveFull</li>
    <li>texliveFull.info</li>
    <li>texliveFull.man</li>
    <li>texliveGUST</li>
    <li>texliveGUST.info</li>
    <li>texliveGUST.man</li>
    <li>texliveInfraOnly</li>
    <li>texliveInfraOnly.info</li>
    <li>texliveInfraOnly.man</li>
    <li>texliveMedium</li>
    <li>texliveMedium.info</li>
    <li>texliveMedium.man</li>
    <li>texliveMinimal</li>
    <li>texliveMinimal.info</li>
    <li>texliveMinimal.man</li>
    <li>texliveSmall</li>
    <li>texliveSmall.info</li>
    <li>texliveSmall.man</li>
    <li>texliveTeTeX</li>
    <li>texliveTeTeX.info</li>
    <li>texliveTeTeX.man</li>
    <li>texmacs</li>
    <li>vimPlugins.openscad-nvim</li>
    <li>wayback-machine-archiver</li>
    <li>wayback-machine-archiver.dist</li>
    <li>zathura</li>
  </ul>
</details>

veprbl commented:
Nothing TeX related, the actual failures are: `recoll` (something about x11), `python311Packages.behave`, `sympow`

xworld21 reviewed (changes requested):
Looks great. The osda package change is a bit worrying – I think we must either trust TeX Live and Debian and include it, or remove it altogether. I made other minor suggestions.

In the meanwhile, I am trying to run nixpkgs-review on Intel Darwin but it will take a while (and the Cambridge CTAN mirror keeps timing out... not a good time for rebuilding TeX Live in the UK!).

An additional discovery: texliveInfraOnly now depends on LuaTeX, because rungs is a Lua script. That kind of ruins the core/core-big distinction. Something to keep in mind for the next refactor.

PS: I am surprised OfBorg shows such a small rebuild. Good for us anyway, we can merge directly to master.

PPS: the first commit message says '2023' instead of '2024'!

xworld21 commented on pkgs/tools/typesetting/tex/texlive/tlpdb-overrides.nix:
```diff
@@ -337,7 +339,9 @@ in lib.recursiveUpdate orig rec {
-  xdvi.deps = (orig.xdvi.deps or []) ++  [ "metafont" ];
+  xdvi.deps = (orig.xdvi.deps or [ ]) ++ [ "metafont" ];
+
+  mltex.deps = (orig.mltex.deps or [ ]) ++ [ "pdftex" ];
```
Forgot to ask: why is this appearing now? (I'd check myself, but I am struggling to build due to mirror issues.)

xworld21 replied:
Found the reason. It's a format built with pdftex. I guess it used to depend transitively on pdftex, but now it does not.

xworld21 commented:
XeTeX on Darwin is now compiled against Core Text/Cocoa, not fontconfig (possibly after rebasing against the latest master). This means the fontconfig file from #228196 does not work on macOS anymore, and the corresponding opentype-fonts test fails. We need to improve the wrapper somehow.

veprbl commented on pkgs/tools/typesetting/tex/texlive/tlpdb-overrides.nix:
```diff
@@ -354,7 +358,22 @@ in lib.recursiveUpdate orig rec {
+    "unfree"
+  ];
+  # hence we can't include this by default
+  collection-publishers.deps = builtins.filter (dep: dep != "osda") orig.collection-publishers.deps;
```
Looks like we are already doing it right for `texlive.osda.pkgs`:
```
       error: Package ‘osda-1.4.0-tex’ in pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix:70 has an unfree license (‘lppl13c unfree’), refusing to evaluate.
```

xworld21 commented on pkgs/tools/typesetting/tex/texlive/bin.nix:
The following addition to `core-big.preConfigure` fixes the problems with the opentype-fonts.xelatex test on Darwin.
```nix
# force XeTeX to use fontconfig instead of Core Text so that fonts can be made available via FONTCONFIG_FILE
+ lib.optionalString stdenv.isDarwin ''
  export kpse_cv_have_ApplicationServices=no
''
```

veprbl replied:
I haven't checked, but I'm guessing that the new behavior could be caused by the new SDK (possibly now part of stdenv?). Perhaps, we could just leave it as-is for this PR.

In general, it's probably not such a bad user experience to let users use their system fonts. We could provide an option and document it to switch between the two options too.

xworld21 replied:
Yes, this must be the new SDK. I rebased your PR as soon as I heard it went live and found the issue that way.

Fontconfig already picks up system fonts (assuming makeFontsConf is still fully functional, there were some accidents there apparently!) so I figure we should avoid the breaking change for now, this close to release, even though it requires a hack (the variable tricks configure into thinking the test result is already in the cache).

The ideal solution is to find an equivalent of `FONTCONFIG_FILE` for Cocoa, of course.

veprbl replied:
If this is already a problem on master, we can work on this in a separate PR?

xworld21 commented on pkgs/test/texlive/default.nix:
```diff
@@ -381,16 +381,16 @@ rec {
+        "kpsewhere" "latex-git-log" "ltxfileinfo" "mendex" "pdflatexpicscale" "perltex" "pn2pdf" "psbook" "psnup" "psresize" "purifyeps"
         "simpdftex" "tex2xindy" "texluac" "texluajitc" "upmendex" "urlbst" "yplan" ];
-      shortHelp = [ "adhocfilelist" "authorindex" "bbl2bib" "bibdoiadd" "bibmradd" "biburl2doi" "bibzbladd" "ctanupload"
+      shortHelp = [ "adhocfilelist" "authorindex" "bbl2bib" "bibdoiadd" "bibmradd" "biburl2doi" "bibzbladd" "bookshelf-mkfontsel" "ctanupload"
```
The better treatment for `bookshelf-listallfonts` is the same as `bookshelf-mkfontsel`:
```suggestion
      shortHelp = [ "adhocfilelist" "authorindex" "bbl2bib" "bibdoiadd" "bibmradd" "biburl2doi" "bibzbladd" "bookshelf-listallfonts" "bookshelf-mkfontsel" "ctanupload"
```
Likewise for `ignoreExitCode`. That way it does not try to call `luaotfload-tool`.

veprbl replied:
Did you test that? IIRC when I tried that, it would not work (no such option, or something). If command runs without the arguments and correctly checks for its dependency `luaotfload-tool` that's a better test, right?

xworld21 replied:
Yes, `bookshelf-listallfonts -h` seems to work fine here. If you wish to test that it works with `luaotfload-tool`, then `needScheme` is more appropriate as it will test the wrapped binary with the dependencies in path, but the `binaries` test doesn't require that you also check functionality (there are too many binaries!).

veprbl replied:
Done. Also had to add to `ignoreExitCode`.

emilazy commented:
This is going to be a breaking change for some workflows, right? We’re meant to be in the feature freeze period, but I guess holding off until 25.05 might be too long to go without a TeX Live update?

veprbl commented:
> This is going to be a breaking change for some workflows, right? We’re meant to be in the feature freeze period, but I guess holding off until 25.05 might be too long to go without a TeX Live update?

There is not supposed to be a breaking change. However, this is meant to be merged after 25.05 branch-off.

veprbl commented on pkgs/tools/typesetting/tex/texlive/bin.nix:
```diff
@@ -38,12 +38,19 @@ let
-              "ftp://tug.ctan.org/pub/tex/historic/systems/texlive/${year}/texlive-${year}0313-source.tar.xz"
-      ];
-      hash = "sha256-OHiqDh7QMBwFOw4u5OmtmZxEE0X0iC55vdHI9M6eebk=";
+    # FIXME revert to official tarballs for TeX-Live 2025
```
It's probably okay to just use GitHub.

nadir-ishiguro commented:
Thank you!
